### ISSUE TITLE
need help to build a dataset from local numpy data

### ISSUE THREAD

**Issue by CharlieLi2S** [tensorflow/datasets#5330]
**What I need help with / What I was wondering**
I intended to build a dataset modifying language_table_sim by embedding the natural language instructions of the original dataset. Here's what I've done: I downloaded language_table_sim and loaded and saved the data as numpy files:
> data
├── train
│ ├── episode_0
│ ├── episode_1
│ ├── …
├── val
│ ├── episode_100
│ ├── episode_101
│ ├── …
├── test
│ ├── episode_201
│ ├── episode_202
│ ├── …
————————————————

then I write the builder script and placed it under the same dir with data
I tried to build the dataset by commanding
> tfds build --data_dir \home\ds

while the error occurs:
>2024-03-21 04:09:22.381563: W tensorflow/tsl/platform/cloud/google_auth_provider.cc:184] All attempts to get a Google authentication bearer token failed, returning an empty token. Retrieving token from files failed with "NOT_FOUND: Could not locate the credentials file.". Retrieving token from GCE failed with "FAILED_PRECONDITION: Error executing an HTTP request: libcurl code 6 meaning 'Couldn't resolve host name', error details: Could not resolve host: metadata.google.internal".

it seems that the builder is connecting to google storage, but I belive I don't need that, because all the data are local
I've searched related issues such as https://github.com/tensorflow/datasets/issues/5194#issue-2043792034 but I can't fix it so I really appreciate that if you can help me

here's my builder script:
"""language_table_use_dataset_builder.py"""

from typing import Iterator, Tuple, Any

import glob
import numpy as np
import tensorflow as tf
import tensorflow_datasets as tfds
import tensorflow_hub as hub

class LanguageTableUse(tfds.core.GeneratorBasedBuilder):
    """DatasetBuilder for example dataset."""

    VERSION = tfds.core.Version('1.0.0')
    RELEASE_NOTES = {
      '1.0.0': 'Initial release.',
    }

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self._embed = hub.load("/home/universal_sentence_encoder")

    def _info(self) -> tfds.core.DatasetInfo:
        """Dataset metadata (homepage, citation,...)."""
        return self.dataset_info_from_configs(
            features=tfds.features.FeaturesDict({
                'steps': tfds.features.Dataset({
                    'observation': tfds.features.FeaturesDict({
                        'rgb': tfds.features.Image(
                            shape=(360, 640, 3),
                            dtype=np.uint8,
                            doc='RGB observation.',
                        ),
                        'effector_target_translation': tfds.features.Tensor(
                            shape=(2,),
                            dtype=np.float32,
                            doc='robot effector target,like x,y in the 2-D dimension',
                        ),
                        'effector_translation': tfds.features.Tensor(
                            shape=(2,),
                            dtype=np.float32,
                            doc='robot effector state,like x,y in the 2-D dimension',
                        ),
                        'instruction': tfds.features.Tensor(
                            shape=(512,),
                            dtype=np.float32,
                            doc='universial sentence embedding instruction',
                        ),
                    }),
                    'action': tfds.features.Tensor(
                        shape=(2,),
                        dtype=np.float32,
                        doc='Robot action',
                    ),
                    'reward': tfds.features.Scalar(
                        dtype=np.float32,
                        doc='Reward if provided, 1 on final step for demos.'
                    ),
                    'is_first': tfds.features.Scalar(
                        dtype=np.bool_,
                        doc='True on first step of the episode.'
                    ),
                    'is_last': tfds.features.Scalar(
                        dtype=np.bool_,
                        doc='True on last step of the episode.'
                    ),
                    'is_terminal': tfds.features.Scalar(
                        dtype=np.bool_,
                        doc='True on last step of the episode if it is a terminal step, True for demos.'
                    ),
                }),
                'episode_metadata': tfds.features.FeaturesDict({
                    'file_path': tfds.features.Text(
                        doc='Path to the original data file.'
                    ),
                }),
            }))

    def _split_generators(self, dl_manager: tfds.download.DownloadManager):
        """Define data splits."""
        return {
            'train': self._generate_examples(path='data/train/episode_*.npy'),
            'val': self._generate_examples(path='data/val/episode_*.npy'),
	    'test': self._generate_examples(path='data/test/episode_*.npy'),
        }

    def _generate_examples(self, path) -> Iterator[Tuple[str, Any]]:
        """Generator of examples for each split."""

        def _parse_example(episode_path):
            # load raw data --> this should change for your dataset
            with open(episode_path,'rb') as file:
                data = np.load(file, allow_pickle=True)     # this is a list of dicts in our case

            def decode_inst(inst):
                return bytes(inst[np.where(inst != 0)].tolist()).decode("utf-8")

            # assemble episode --> here we're assuming demos so we set reward to 1 at the end
            episode = []
            for i, step in enumerate(data):
                # compute Kona language embedding
                language_embedding = self._embed([decode_inst(np.array(step['instruction']))])[0].numpy()

                episode.append({
                    'observation': {
                        'rgb': step['rgb'],
                        'effector_target_translation': step['effector_target_translation'],
                        'effector_translation': step['effector_translation'],
                        'instruction': language_embedding,
                    },
                    'action': step['action'],
                    'reward': step['reward'],
                    'is_first': step['is_first'],
                    'is_last': step['is_last'],
                    'is_terminal': step['is_terminal'],
                })

            # create output data sample
            sample = {
                'steps': episode,
                'episode_metadata': {
                    'file_path': episode_path
                }
            }

            # if you want to skip an example for whatever reason, simply return None
            return episode_path, sample

        # create list of all examples
        episode_paths = glob.glob(path)

        # for smallish datasets, use single-thread parsing
        for sample in episode_paths:
            yield _parse_example(sample)

        # for large datasets use beam to parallelize data parsing (this will have initialization overhead)
        # beam = tfds.core.lazy_imports.apache_beam
        # return (
        #         beam.Create(episode_paths)
        #         | beam.Map(_parse_example)
        # )

**Environment information**
I was running it inside a docker container, the image is tensorflow 2.14-gpu, and I've install several packages myself
* Operating System: Ubuntu 22.04.2 LTS
* Python version: Python 3.11.0rc1
* `tensorflow-datasets`/`tfds-nightly` version: tensorflow-datasets 4.9.4
* `tensorflow`/`tensorflow-gpu`/`tf-nightly`/`tf-nightly-gpu` version: tensorflow 2.14.0
> Package                      Version
---------------------------- ---------------------
absl-py                      1.4.0
apache-beam                  2.54.0
array-record                 0.5.0
astunparse                   1.6.3
attrs                        23.2.0
blinker                      1.4
cachetools                   5.3.1
certifi                      2023.7.22
charset-normalizer           3.2.0
chex                         0.1.85
click                        8.1.7
cloudpickle                  2.2.1
clu                          0.0.11
contextlib2                  21.6.0
crcmod                       1.7
cryptography                 3.4.8
dbus-python                  1.2.18
decorator                    5.1.1
dill                         0.3.1.1
distro                       1.7.0
dm-reverb                    0.13.0
dm-tree                      0.1.8
dnspython                    2.6.1
docopt                       0.6.2
etils                        1.7.0
fastavro                     1.9.4
fasteners                    0.19
flatbuffers                  23.5.26
flax                         0.8.1
fsspec                       2024.2.0
gast                         0.5.4
gin-config                   0.5.0
google-auth                  2.23.1
google-auth-oauthlib         1.0.0
google-pasta                 0.2.0
googleapis-common-protos     1.62.0
grpcio                       1.58.0
gym                          0.23.0
gym-notices                  0.0.8
h5py                         3.9.0
hdfs                         2.7.3
httplib2                     0.20.2
idna                         3.4
importlib-metadata           4.6.4
importlib_resources          6.3.2
jax                          0.4.25
jaxlib                       0.4.25+cuda11.cudnn86
jeepney                      0.7.1
Js2Py                        0.74
jsonpickle                   3.0.3
jsonschema                   4.21.1
jsonschema-specifications    2023.12.1
keras                        2.14.0
keyring                      23.5.0
launchpadlib                 1.10.16
lazr.restfulclient           0.14.4
lazr.uri                     1.0.6
libclang                     16.0.6
Markdown                     3.4.4
markdown-it-py               3.0.0
MarkupSafe                   2.1.3
mdurl                        0.1.2
ml-collections               0.1.1
ml-dtypes                    0.2.0
more-itertools               8.10.0
msgpack                      1.0.8
namex                        0.0.7
nest-asyncio                 1.6.0
numpy                        1.24.4
oauthlib                     3.2.0
objsize                      0.7.0
opt-einsum                   3.3.0
optax                        0.2.1
optree                       0.10.0
orbax-checkpoint             0.4.4
orjson                       3.9.15
packaging                    23.1
pillow                       10.2.0
pip                          23.2.1
pipdeptree                   2.16.1
portpicker                   1.6.0
promise                      2.3
proto-plus                   1.23.0
protobuf                     3.20.3
psutil                       5.9.8
pyarrow                      14.0.2
pyarrow-hotfix               0.6
pyasn1                       0.5.0
pyasn1-modules               0.3.0
pybullet                     3.2.6
pydot                        1.4.2
pygame                       2.1.3
Pygments                     2.17.2
PyGObject                    3.42.1
pyjsparser                   2.7.1
PyJWT                        2.3.0
pymongo                      4.6.2
pyparsing                    2.4.7
python-apt                   2.4.0+ubuntu2
python-dateutil              2.9.0.post0
pytz                         2024.1
PyYAML                       6.0.1
referencing                  0.33.0
regex                        2023.12.25
requests                     2.31.0
requests-oauthlib            1.3.1
rich                         13.7.1
rlds                         0.1.8
rpds-py                      0.18.0
rsa                          4.9
scipy                        1.12.0
SecretStorage                3.3.1
setuptools                   68.2.2
six                          1.16.0
tensorboard                  2.14.0
tensorboard-data-server      0.7.1
tensorflow                   2.14.0
tensorflow-datasets          4.9.4
tensorflow-estimator         2.14.0
tensorflow-hub               0.14.0
tensorflow-io-gcs-filesystem 0.34.0
tensorflow-metadata          1.14.0
tensorflow-probability       0.22.1
tensorflow-serving-api       2.14.0
tensorstore                  0.1.40
termcolor                    2.3.0
tf-agents                    0.18.0
tf-slim                      1.1.0
toml                         0.10.2
toolz                        0.12.1
tqdm                         4.66.2
typing_extensions            4.5.0
tzlocal                      5.2
urllib3                      2.0.5
wadllib                      1.3.6
Werkzeug                     2.3.7
wheel                        0.41.2
wrapt                        1.14.1
zipp                         1.0.0
zstandard                    0.22.0

**Comment by rishusam** [comment]
The error message you encountered suggests that TensorFlow is trying to authenticate with Google Cloud services to retrieve authentication tokens, but it's unable to do so because it's running in an environment where it can't access the necessary credentials.

Since you're working with a local dataset and don't need to interact with Google Cloud storage, you can disable the Google authentication by setting the environment variable `GOOGLE_APPLICATION_CREDENTIALS` to an empty string before running your script.

Here's how you can modify your script to disable Google authentication:

```python
import os

# Disable Google Cloud authentication
os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = ''

# Now import the required modules
import glob
import numpy as np
import tensorflow as tf
import tensorflow_datasets as tfds
import tensorflow_hub as hub

# Your existing code follows...
```

Adding this code at the beginning of your script will prevent TensorFlow from attempting to authenticate with Google Cloud services, and it should resolve the error you encountered.

Additionally, make sure that the file paths you're providing in your script (`'/home/universal_sentence_encoder'`) are correct and accessible within your Docker container environment.

Once you've made these modifications, try running your script again:

```
python language_table_use_dataset_builder.py
```

This should allow your script to build the dataset without encountering authentication errors. If you encounter any further issues, please let me know, and I'll be happy to assist you further.

**Comment by CharlieLi2S** [comment]
> The error message you encountered suggests that TensorFlow is trying to authenticate with Google Cloud services to retrieve authentication tokens, but it's unable to do so because it's running in an environment where it can't access the necessary credentials.
> 
> Since you're working with a local dataset and don't need to interact with Google Cloud storage, you can disable the Google authentication by setting the environment variable `GOOGLE_APPLICATION_CREDENTIALS` to an empty string before running your script.
> 
> Here's how you can modify your script to disable Google authentication:
> 
> ```python
> import os
> 
> # Disable Google Cloud authentication
> os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = ''
> 
> # Now import the required modules
> import glob
> import numpy as np
> import tensorflow as tf
> import tensorflow_datasets as tfds
> import tensorflow_hub as hub
> 
> # Your existing code follows...
> ```
> 
> Adding this code at the beginning of your script will prevent TensorFlow from attempting to authenticate with Google Cloud services, and it should resolve the error you encountered.
> 
> Additionally, make sure that the file paths you're providing in your script (`'/home/universal_sentence_encoder'`) are correct and accessible within your Docker container environment.
> 
> Once you've made these modifications, try running your script again:
> 
> ```
> python language_table_use_dataset_builder.py
> ```
> 
> This should allow your script to build the dataset without encountering authentication errors. If you encounter any further issues, please let me know, and I'll be happy to assist you further.

Thanks for your suggestion, but it seems that the issue is still remaining

**Comment by fylux** [comment]
Hi,

Can you confirm that this message is related to your dataset preparation not working? Seems like it could be just a warning that shouldn't prevent the script from executing (see https://github.com/tensorflow/datasets/issues/2761)

You can also try using:
```
tfds.core.utils.gcs_utils._is_gcs_disabled = True
os.environ['NO_GCE_CHECK'] = 'true'
```
To avoid the error message

Let us know if it works.

**Comment by CharlieLi2S** [comment]
> Hi,
> 
> Can you confirm that this message is related to your dataset preparation not working? Seems like it could be just a warning that shouldn't prevent the script from executing (see #2761)
> 
> You can also try using:
> 
> ```
> tfds.core.utils.gcs_utils._is_gcs_disabled = True
> os.environ['NO_GCE_CHECK'] = 'true'
> ```
> 
> To avoid the error message
> 
> Let us know if it works.

it works, thanks
